File: Leetcode/dynamic_programming/746_MinCostClimbingStair/Solution.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        n = len(cost)
        if n == 1: return cost[0]
        for i in range(2, n):
            cost[i] += min(cost[i-1], cost[i-2])
        return min(cost[-2], cost[-1])

File: Leetcode/dynamic_programming/746_MinCostClimbingStair/Solution.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        if(n==1) return cost[0];
        int[] dp = new int[n+1];
        dp[0] = cost[0]; dp[1] = cost[1];
        for(int i = 2; i<=n; ++i){
            dp[i] = Math.min(dp[i-1], dp[i-2]);
            if(i<n){
                dp[i] += cost[i];
            }
        }
        return dp[n];
    }
}
